platform/linuxbsd/wayland/wayland_thread.cpp

```diff
@@ -1589,7 +1589,20 @@ void WaylandThread::_wl_seat_on_capabilities(void *data, struct wl_seat *wl_seat
 	ERR_FAIL_NULL(ss);
 
 	// TODO: Handle touch.
+	if (capabilities & WL_SEAT_CAPABILITY_TOUCH) {
+		if (!ss->wl_touch) {
+			ss->cursor_surface = wl_compositor_create_surface(ss->registry->wl_compositor);
+			wl_surface_commit(ss->cursor_surface);
 
+			ss->wl_touch = wl_seat_get_touch(wl_seat);
+			wl_touch_add_listener(ss->wl_touch, &wl_touch_listener, ss);
+		}
+	} else {
+		if (ss->wl_touch) {
+			wl_touch_destroy(ss->wl_touch);
```

Comment:
Needs uninitialization for the touch data.

Also you need to uninitialize the touch object into `WaylandThread::destroy`, like with the rest of the input objects (yes, this is annoying, I plan on removing this duplication)